Sources/Onboarding/ConsentView.swift

```diff
@@ -106,6 +106,34 @@ public struct ConsentView<ContentView: View, Action: View>: View {
             }
         )
     }
+
+    /// Creates a ``ConsentView`` with a provided action view using  an``OnboardingActionsView`` and renders HTML in a web view.
+    /// - Parameters:
+    ///   - header: The header view will be displayed above the html content.
+    ///   - asyncHTML: The html content provided as an UTF8 encoded `Data` instance that can be provided asynchronously.
+    ///   - footer: The footer view will be displayed above the html content.
+    ///   - action: The action that should be performed once the consent has been given.
+    public init(
+        @ViewBuilder header: () -> (some View) = { EmptyView() },
+        asyncHTML: @escaping () async -> Data,
```

Comment:
I would suggest that we add a `DocumentView` type that can either get an HTML file or a markdown data and has some sort of `viewRepresentation` property of function that just returns an `AnyView` to render on screen (that is then the `HTMLView` or `MarkdownView`).
Then we only have a single initializer and can group that functionality in a distinct abstraction 👍

Looking at the current code that I wrote, I would even argue that it makes sense to move the header and footer views out of the Markdown and `HTMLView` views and put them right in the ConsentView. Then the MD and HTML views only focus on their own content and footers and headers are up to the user of these views to implement or not even use. That might even make is easier implementing the above mentioned functionality as it would just allow you to implement it using a property 🚀